Improve acceptance test happiness

tbg opened the pull request:
- Clean up nodes even on failure
- Set a log level that actually prints things
- Create error log when container fails, even if no LogDir set

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4698)

<!-- Reviewable:end -->

petermattis commented:
LGTM

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/localcluster.go, line 433 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4698#-KBYCGR1Ug8qWK9CnchT:-KBYCGR1Ug8qWK9CnchU:-1263384833) ([raw file](https://github.com/cockroachdb/cockroach/blob/267586720fbd129703d7f8a6298d259d77d1e3b0/acceptance/cluster/localcluster.go#L433)):*
FYI, I think this is equivalent to `--alsologtostderr`. Fine either way.

---

*[util/testing.go, line 45 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4698#-KBYCtH_Zp9qSKWZqsB8:-KBYCtH_Zp9qSKWZqsB9:-125685905) ([raw file](https://github.com/cockroachdb/cockroach/blob/267586720fbd129703d7f8a6298d259d77d1e3b0/util/testing.go#L45)):*
The receiver doesn't need to be a pointer. And then `Tester` would be defined as `Tester = panicTesterImpl{}`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4698#-:-KBYD4O9gyef8sLkRHxC:676834233)*
<!-- Sent from Reviewable.io -->

tbg committed:
Improve acceptance test happiness

- Clean up nodes even on failure
- Set a log level that actually prints things
- Create error log when container fails, even if no LogDir set
